src/Middleware/Rewrite/src/RedirectToRootFromWwwRule.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Rewrite.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Rewrite
+{
+    internal class RedirectToRootFromWwwRule : IRule
+    {
+        public readonly int _statusCode;
+        public readonly string[] _domains;
+
+        public RedirectToRootFromWwwRule(int statusCode)
+        {
+            _statusCode = statusCode;
+        }
+
+        public RedirectToRootFromWwwRule(int statusCode, params string[] domains)
+        {
+            if (domains == null)
+            {
+                throw new ArgumentNullException(nameof(domains));
+            }
+
+            if (domains.Length < 1)
+            {
+                throw new ArgumentException(nameof(domains));
+            }
+
+            _domains = domains;
+            _statusCode = statusCode;
+        }
+
+        public virtual void ApplyRule(RewriteContext context)
+        {
+            var req = context.HttpContext.Request;
+
+            if (req.Host.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (!req.Host.Value.StartsWith("www.", StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (_domains != null)
```

Comment:
`_domains` is never null since you are throwing an exception if so in constrictor.